extensions/kubernetes-client/deployment-internal/src/main/java/io/quarkus/kubernetes/client/deployment/KubernetesClientBuildStep.java

```diff
@@ -15,4 +21,23 @@ public class KubernetesClientBuildStep {
     public KubernetesClientBuildItem process(TlsConfig tlsConfig) {
         return new KubernetesClientBuildItem(createClient(buildConfig, tlsConfig));
     }
+
+    @BuildStep(onlyIfNot = IsStaticLoggerEnabled.class)
+    public void produceLoggingCategories(BuildProducer<LogCategoryBuildItem> categories) {
+        categories.produce(new LogCategoryBuildItem("okhttp3.OkHttpClient", Level.WARN));
+    }
+
+    public static class IsStaticLoggerEnabled implements BooleanSupplier {
+
+        final KubernetesClientBuildConfig config;
+
+        public IsStaticLoggerEnabled(KubernetesClientBuildConfig config) {
+            this.config = config;
+        }
+
+        @Override
+        public boolean getAsBoolean() {
+            return config.enabled;
+        }
+    }
```

Comment:
Same question as Georgios but also it would be far simpler if you just injected the config in the build step and added a if. You could get rid of all this.

Reply:
It will be probably better, will try that, thanks! :)